### ISSUE TITLE
Add wasmer-compiler-cranelift

### ISSUE THREAD

**Issue by ibdafna** [conda-forge/wasmer-feedstock#16]
- [x] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list

```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

```
</details>

Embedding the Wasmer runtime in other contexts requires a compiler backend - cranelift is one of the three (llvm and singlepass are the two other ones) and is the only one with reasonable speed that can be used in Windows. It would be awesome if can can add it - happy to do so myself with some guidance 🙏  Thank you! 😄

**Comment by bollwyvl** [comment]
Yes, there are _many_ other build options, and the matrix of what works (well) with what else is very complicated, and likely to change over time.

- https://github.com/wasmerio/wasmer/blob/master/Makefile#L12-L18
- https://github.com/wasmerio/wasmer/blob/master/PACKAGING.md

I'd like to see us get to where we had:

- the embeddable part
- the smallest/lightest CLI-enabled package available that most packages that just needed if they wanted to run some pre-compiled wasm
- a reasonable set of options, preferably without having to resort to too many weird things like variants, etc. for the different backends
  - this could be multiple build numbers, etc.
- robust tests
  - probably not the full upstream suite

Also (and as separate feedstocks) we would also want the language runtimes, starting with https://github.com/wasmerio/wasmer-python, probably.

But, as I said, there's a lot. If you step back through the history of `meta.yaml`, there were some additional backend tests in, but I had taken them out... if I personally was making a real run of it, due to the matrix of things, i'd probably start with the tests: 
- pytest
- a bunch of fixtures to get an idea of what we don't have
